apis/python/src/tiledb/vector_search/module.cc

```diff
@@ -165,16 +165,44 @@ static void declare_qv_query_heap_infinite_ram(
          std::vector<Id_Type>& ids,
          size_t nprobe,
          size_t k_nn,
-         size_t nthreads) -> py::tuple {
+         size_t nthreads,
+         DistanceMetric distance_metric = DistanceMetric::L2) -> py::tuple {
         auto mat = ColMajorPartitionedMatrixWrapper<T, Id_Type, Id_Type>(
             parts, ids, indices);
 
         auto top_centroids = detail::ivf::ivf_top_centroids(
             centroids, query_vectors, nprobe, nthreads);
-        auto r = detail::ivf::qv_query_heap_infinite_ram(
-            top_centroids, mat, query_vectors, nprobe, k_nn, nthreads);
-
-        return make_python_pair(std::move(r));
+        if (distance_metric == DistanceMetric::L2) {
+          auto r = detail::ivf::qv_query_heap_infinite_ram(
+              top_centroids,
+              mat,
+              query_vectors,
+              nprobe,
+              k_nn,
+              nthreads,
+              sum_of_squares_distance{});
+          return make_python_pair(std::move(r));
+        } else if (distance_metric == DistanceMetric::INNER_PRODUCT) {
+          auto r = detail::ivf::qv_query_heap_infinite_ram(
+              top_centroids,
+              mat,
+              query_vectors,
+              nprobe,
+              k_nn,
+              nthreads,
+              inner_product_distance{});
+          return make_python_pair(std::move(r));
+        } else if (distance_metric == DistanceMetric::COSINE) {
+          auto r = detail::ivf::qv_query_heap_infinite_ram(
+              top_centroids,
+              mat,
+              query_vectors,
+              nprobe,
+              k_nn,
+              nthreads,
+              cosine_distance_normalized{});
+          return make_python_pair(std::move(r));
+        }
       },
```

Comment:
Heads up I'm seeing lots of new warnings like:
```
  /Users/parismorgan/repo/TileDB-Vector-Search/apis/python/src/tiledb/vector_search/module.cc:820:3: note: in instantiation of function template specialization '(anonymous namespace)::declare_vq_query_heap_pyarray<signed char, unsigned long long>' requested here
    declare_vq_query_heap_pyarray<int8_t>(m, "i8");
    ^
  /Users/parismorgan/repo/TileDB-Vector-Search/apis/python/src/tiledb/vector_search/module.cc:626:7: warning: non-void lambda does not return a value in all control paths [-Wreturn-type]
        });
```
when I build - seems  nice to add an `else()` here and to the rest of the functions that throws.